src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechRecognizer.java

```diff
@@ -112,82 +117,10 @@
 
   public AbstractSpeechRecognizer(String n, String id) {

```

Comment:
Please add the "abstract" key word here.  just because the class starts with the word "Abstract" doesn't mean it's abstract.  classes defined as abstract should be   "public abstract class"

Reply:
I guess your looking at the constructor .. the class is decorated with abstract

Reply:
public AbstractSpeechRecognizer(String n, String id) 
should read
public abstract AbstractSpeechRecognizer(String n, String id)

o/w it's not actaully abstract.

Reply:
You are incorrect.
https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html

_"An abstract class is a class that is declared abstract—it may or may not include abstract methods. Abstract classes cannot be instantiated, but they can be subclassed."_